### PR TITLE
Ryt

### PR DESCRIPTION
**Pull request by declmal**
Thanks for contributing to TVM!   Please refer to guideline https://docs.tvm.ai/contribute/ for useful information and tips. After the pull request is submitted, please request code reviews from [Reviewers](https://github.com/dmlc/tvm/blob/master/CONTRIBUTORS.md#reviewers).

### CODE REVIEW

File: cvm/quantization/tfm_ops.py [hunk 1]
```diff
@@ -173,6 +207,53 @@ def validate(self, op, **kwargs):
...
+            conv_name = name.replace('pool', 'pool_conv')
+            W_name = conv_name + '_weight'
+            assert W_name not in graph
+            W_shape = (in_channel, 1, *kernel)
+            graph[W_name] = W = mx.sym.var(W_name, shape=W_shape)
+            params[W_name] = nd.full(shape=W_shape, val=(1/np.product(kernel)))
+            op = mx.sym.Convolution(X, W, **conv_attr, name=conv_name)
+        else:
+            assert pool_type == 'max', "Unsupported Pooling \
+                    %s(%s, pool_type=%s)"%(Pooling.op_name, name, pool_type)
```
Comment (wlt-cortex) [inline cvm/quantization/tfm_ops.py]: The pool_type has been validated in `validate` pass.

**Review by wlt-cortex — CHANGES_REQUESTED** [review]

File: cvm/quantization/tfm_ops.py [hunk 2]
```diff
@@ -132,6 +143,25 @@ def calculate_ops(self, op, **kwargs):
 @register_pass("fuse_transpose")
 @register_transformer("FullyConnected")
 class FullyConnected(Transformer):
+    def rewrite(self, op, **kwargs):
+        #TODO: matrix decomposition
```
Comment (wlt-cortex) [inline cvm/quantization/tfm_ops.py]: Matrix decomposition

File: cvm/quantization/tfm_ops.py (hunk 1 above)
Comment (declmal) [reply cvm/quantization/tfm_ops.py]: updated

File: cvm/quantization/mrt.py [hunk 3]
```diff
@@ -553,12 +553,14 @@ def _sym_calibrate(self, ctx, lambd):
         order, deps = topo_sort(self.sym, logger=self._lgr, with_deps=True)
         old_ths = self.th_dict if self.th_dict else {}
         self.th_dict, out_cache = {}, {}
+        ts = set()
```
Comment (wlt-cortex) [inline cvm/quantization/mrt.py]: revert debug code

File: cvm/quantization/mrt.py [hunk 4]
```diff
@@ -553,12 +553,14 @@ def _sym_calibrate(self, ctx, lambd):
...
         for sym in order:
             name, op_name = sym.attr('name'), sym.attr('op_name')
             attr, childs = sym.list_attr(), sym_iter(sym.get_children())
             if op_name == 'null':
                 out = self._datas[name] if name in self.ins_ext \
                       else self.prm[name]
+                ts.add(len(sym))
```
Comment (wlt-cortex) [inline cvm/quantization/mrt.py]: revert

File: cvm/quantization/mrt.py [hunk 5]
```diff
@@ -584,6 +586,7 @@ def _sym_calibrate(self, ctx, lambd):
                         name, [o.shape for o in out], self.th_dict[name])
 
         out_cache.clear()
+        print(ts)
```
Comment (wlt-cortex) [inline cvm/quantization/mrt.py]: revert

**Review by wlt-cortex — CHANGES_REQUESTED** [review]

File: cvm/quantization/mrt.py (hunk 3 above)
Comment (declmal) [reply cvm/quantization/mrt.py]: reverted

File: cvm/quantization/mrt.py (hunk 4 above)
Comment (declmal) [reply cvm/quantization/mrt.py]: reverted

File: cvm/quantization/mrt.py (hunk 5 above)
Comment (declmal) [reply cvm/quantization/mrt.py]: reverted

**Pull request merged**